web/components/contract/swipe-card.tsx

```diff
@@ -0,0 +1,202 @@
+import clsx from 'clsx'
+import {
+  ArrowLeftIcon,
+  ArrowRightIcon,
+  MinusIcon,
+  PlusIcon,
+} from '@heroicons/react/solid'
+
+import { getOutcomeProbabilityAfterBet } from 'common/calculate'
+import { BinaryContract, Contract } from 'common/contract'
+import { getBinaryProb } from 'common/contract-details'
+import { formatMoney, formatPercent } from 'common/util/format'
+import { richTextToString } from 'common/util/parse'
+import { memo } from 'react'
+import { contractPath } from 'web/lib/firebase/contracts'
+import { fromNow } from 'web/lib/util/time'
+import { Avatar } from '../widgets/avatar'
+import { SiteLink } from '../widgets/site-link'
+import { Row } from '../layout/row'
+import { NoLabel, YesLabel } from '../outcome-label'
+import { Col } from '../layout/col'
+import { useContract } from 'web/hooks/use-contracts'
+
+const betTapAdd = 10
+
+export const SwipeCard = memo(
+  (props: {
+    contract: BinaryContract
+    amount: number
+    setAmount: (amount: number) => void
+    swipeDirection: 'YES' | 'NO' | undefined
+    className?: string
+  }) => {
+    const { amount, setAmount, swipeDirection, className } = props
+    const contract = (useContract(props.contract.id) ??
+      props.contract) as BinaryContract
+    const { question, description, coverImageUrl } = contract
+
+    const addMoney = () => setAmount(amount + betTapAdd)
+
+    const subMoney = () => {
+      if (amount <= betTapAdd) {
+      } else {
+        setAmount(amount - betTapAdd)
+      }
+    }
+
+    const image =
+      coverImageUrl ??
+      `https://picsum.photos/id/${parseInt(contract.id, 36) % 1000}/512`
+
+    return (
+      <Col
+        className={clsx(className, 'user-select-none relative flex h-full')}
+        onClick={(e) => e.preventDefault()}
+      >
+        {/* background */}
+        <div className="flex h-full flex-col bg-black">
+          <div className="relative mb-24 grow">
+            <img
+              src={image}
+              alt=""
+              className="h-full object-cover"
+              style={{ filter: 'brightness(0.60)' }}
+            />
+            <div className="absolute inset-0 bg-gradient-to-t from-black via-transparent" />
+          </div>
+        </div>
+
+        {/* content */}
+        <div className="absolute inset-0 flex select-none flex-col gap-4">
+          <CornerDetails contract={contract} />
+          <SiteLink
+            className="line-clamp-6 mx-8 mt-auto mb-4 text-2xl text-white"
+            href={contractPath(contract)}
+            followsLinkClass
+          >
+            {question}
+          </SiteLink>
+          <Percent
+            contract={contract}
+            amount={amount}
+            outcome={swipeDirection}
+          />
+          {/* TODO: use editor excluding widgets */}
```

Comment:
well, we do want formatted text and mentions but we don't want to show images, tweets, embeds, videos

probably we just use the rich text parser, but change the extensions such that the block level things don't render at all?